trie-db/test/src/triedb.rs

```diff
@@ -644,6 +644,80 @@ fn test_recorder_with_cache_get_hash_internal<T: TrieLayout>() {
 	}
 }
 
+#[test]
+fn test_merkle_value() {
```

Comment:
The test should also check that the functions works correctly for keys that aren't acuatlly in the tree.
E.g. for a tree with nodes ["AAA"] it should return a closest descendant for the key "AA" but not for "BB". For the tree with nodes ["AAAA", "AABA] closes descendant for the key "AA" should be the branch node, etc. 
Also there's no need to make changes to the test trie after it is created really.


Also, you don't need to

Reply:
> Also, you don't need to

Argh, the suspense :D